app/Http/Controllers/PromotionController.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+use App\Promotion;
+use Validator;
+
+class PromotionController extends Controller
+{
+    /**
+     * Display all promotions of the all bussiness.
+     *
+     * @return Response
+     */
+    public function index()
+    {
+        $data = Promotion::all();
+        return view('promotion.list', ['data' => $data]);
+    }
+
+    /**
+     * Display form add promotion.
+     *
+     * @return Response
+     */
+    public function showAddForm()
```

Comment:
[rank C] I think this function name is wrong. We need follow function name as resource route